core/src/main/java/com/linecorp/armeria/common/HttpResponse.java

```diff
@@ -937,4 +938,17 @@ default HttpResponse recover(Function<? super Throwable, ? extends HttpResponse>
         requireNonNull(function, "function");
         return of(new RecoverableStreamMessage<>(this, function, /* allowResuming */ false));
     }
+
+    default HttpResponse recover(Class<? extends Throwable> causeClass,
+                                 Function<? super Throwable, ? extends HttpResponse> function) {
```

Comment:
- Sould we make this type-safer?
```suggestion
    default <T extends Throwable> HttpResponse recover(Class<T> causeClass,
                                 Function<? super T, ? extends HttpResponse> function) {
```
- How about adding the same variant to `StreamMessage.recoverAndResume()`?

Reply:
**Should we make this type-safer?**
- I will try it  :)

**How about adding the same variant to StreamMessage.recoverAndResume()?**
- I checked the StreamMessage.recoverAndResume() method.
Are you referring to the shortcut function of StreamMessage.recoverAndResume()?

Reply:
i agree adding the same variant to StreamMessage.recoverAndResume() i wiil try it :-)